static/js/theme-toggle.js

```diff
@@ -0,0 +1,37 @@
+document.addEventListener("DOMContentLoaded", function () {
+
+// Function to set the theme
+function setTheme(theme) {
+  document.body.setAttribute("data-theme", theme); // Apply to body
+  localStorage.setItem("theme", theme);
+  const themeToggle = document.getElementById("theme-toggle-base");
```

Comment:
I think this should be `theme-toggle`:
```suggestion
  const themeToggle = document.getElementById("theme-toggle");
```